### PR TITLE
dev/core#248 - Fixes advanced search mailings results headers

### PR DESCRIPTION
**Pull request by stesi561**
Removes language from advanced search results mailings column
headers for non-multilingual sites.

Overview
----------------------------------------
Fixes a misalignment of headers for advanced search results when displaying results as mailing.

https://lab.civicrm.org/dev/core/issues/248

Before
----------------------------------------
On sites that are not multilingual the language column is included as a header but not as a result meaning the results and the headers are out of alignment.

After
----------------------------------------
Results are back in alignment with headers.

### CODE REVIEW

**Comment by stesi561** [comment]
There is probably a more elegant way to do this but figured was better to try and minimise the amount of changes to the code.

**Comment by eileenmcnaughton** [comment]
add to whitelist

**Comment by eileenmcnaughton** [comment]
@mlutfy thoughts - seems good to me

File: CRM/Mailing/Selector/Search.php [hunk 1]
```diff
@@ -359,11 +363,17 @@ public function &getColumnHeaders($action = NULL, $output = NULL) {
           'sort' => 'mailing_name',
           'direction' => CRM_Utils_Sort::DONTCARE,
         ],
-        [
+      ];
+
+      // Check to see if languages column should be displayed.
+      if ($isMultiLingual) {
+        $headers[] =  [
```
Comment (seamuslee001) [inline CRM/Mailing/Selector/Search.php]: @stesi561 the style checker expects only 1 space after the = here not 2

**Comment by stesi561** [comment]
This will need testing on a multilingual site.

File: CRM/Mailing/Selector/Search.php [hunk 2]
```diff
@@ -341,8 +341,12 @@ public function getQILL() {
    *   the column headers that need to be displayed
    */
   public function &getColumnHeaders($action = NULL, $output = NULL) {
+
     if (!isset(self::$_columnHeaders)) {
-      self::$_columnHeaders = [
+      $enabledLanguages = CRM_Core_I18n::languages(TRUE);
+      $isMultiLingual = (count($enabledLanguages) > 1);
```
Comment (mlutfy) [inline CRM/Mailing/Selector/Search.php]: Could you replace this with: `CRM_Core_I18n::isMultiLingual()` ?

**Comment by mlutfy** [comment]
I did a bit of testing on the CI-generated site and it does not break anything.

I mean, multilingual is wonky, but I checked some existing multilingual sites and it was broken before.

From the test site, running multilingual:

Capture d’écran de 2019-04-27 16-01-25

This PR does not make it even more broken, and fixes non-multilingual, so it's a good step forward ;)

**Pull request merged**